numba/np/ufunc_db.py

```diff
@@ -583,16 +583,67 @@ def _fill_ufunc_db(ufunc_db):
         'f->f': npyfuncs.np_real_floor_impl,
         'd->d': npyfuncs.np_real_floor_impl,
     }
+    if numpy_version >= (2, 1):
+        ufunc_db[np.floor].update({
+            '?->?': numbers.identity_impl,
+            'b->b': numbers.identity_impl,
+            'B->B': numbers.identity_impl,
+            'h->h': numbers.identity_impl,
+            'H->H': numbers.identity_impl,
+            'i->i': numbers.identity_impl,
+            'I->I': numbers.identity_impl,
+            'l->l': numbers.identity_impl,
+            'L->L': numbers.identity_impl,
+        })
+        if IS_WIN32:
```

Comment:
Why are these masked off for windows (same question for all implementations that are being updated)? e.g. on OSX-arm64 the loops exist (try printing `np.floor.types`).

Reply:
I just tried removing this mask and running the tests on `osx-arm64` and they all pass. I'm not sure why special treatment would be needed for windows, the `q` and `Q` types and loops are defined for other OS?

Reply:
fix confirmed